can/broadcastmanager.py

```diff
@@ -120,18 +136,20 @@ def stop(self):
     def start(self):
         self.stopped = False
         if self.thread is None or not self.thread.is_alive():
-            name = "Cyclic send task for 0x%X" % (self.message.arbitration_id)
+            name = "Cyclic send task for 0x%X" % (self.messages[0].arbitration_id)
             self.thread = threading.Thread(target=self._run, name=name)
             self.thread.daemon = True
             self.thread.start()
 
     def _run(self):
+        msg_index = 0
         while not self.stopped:
             # Prevent calling bus.send from multiple threads
             with self.lock:
                 started = time.time()
                 try:
-                    self.bus.send(self.message)
+                    self.bus.send(self.messages[msg_index])
+                    msg_index = (msg_index + 1) % len(self.messages)
```

Comment:
You could move the msg_index increment outside the lock.

Reply:
Maybe I'm missing something, but why is this lock even necessary? What is this lock intended to protect?

From what I understand, there is a `1:1` mapping between cyclic tasks and threads, so `_run` is essentially running in its own context for each thread. As such, the thread can get preempted, but nothing should be modifying its context state while it is context switched out.

According to `git blame`, you were the one that added this locking, so I thought I'd take the opportunity to ask you :joy:

If this lock is actually protecting against other threads poking into the context, then I believe the increment still needs to be protected by the lock, since Python doesn't provide atomics.

Reply:
I think the lock prevents `Bus.send(...)` from being called concurrently, and not the update of `msg_index`. This is required since `Bus.send(...)` is not guaranteed to be thread safe.

Reply:
Ah, thanks!